scripts/hash-to-file.js

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+/* eslint-disable @typescript-eslint/no-floating-promises */
+const { writeFile } = require("fs/promises");
+const { exec: execCallback } = require("child_process");
+const { promisify } = require("util");
+const { join, resolve } = require("path");
+
+const exec = promisify(execCallback);
+
+(async () => {
+    try {
+        const toPath = process.argv[2] || "";
+
+        const hash = await exec("git rev-parse --short HEAD");
+        const contents = `export const hash = "${hash.stdout.trim()}";\n`;
+
+        await writeFile(resolve(join("./", toPath, "hash.ts")), contents, "utf8");
```

Comment:
I see one problem: this updates a file that is tracked by git, so running a build results in "modified" repository state :/
I think that instead we should put this file together with other generated files in `dist`. Perhaps it shouldn't be importable, but a json or something.

Reply:
Yes, I was thinking about it. Tried to ignore it at first but since it's imported it needs to be there (if not, running via `ts-node` will fail). Also thought about putting it into`dist/` but since the paths may be relative different, e.g. when `sth` is used as submodule, it may not be reliable.

How about putting it as ignored json file in package `src` and then having kind of util which will read from it during runtime? And it will be simple copied to dist during build?

Reply:
What about not importing it, but just reading its contents (only if it's present)?
The thing with ignored file in `src` doesn't seem right to me... and it won't be there before the first build, so you need to handle the case when the file is absent anyway :D

Reply:
I changed it in the meantime trying to make it similar to what you had described - so having ignored json file with version hash which is the read during runtime. It also handles case where the file is absent. Please take a look at - https://github.com/scramjetorg/transform-hub/pull/402/commits/eba5f3276ab1d39589eb6101af5df0dd153cb0db.

Reply:
Yeah, that's more or less what I meant.
It seems that if someone builds one commit, and then runs from source from another commit, `build` will still report the first commit ID. That would be quite confusing. I suggest not to keep this file with sources, but put it directly to `dist`.

Reply:
> It seems that if someone builds one commit, and then runs from source from another commit, `build` will still report the first commit ID. That would be quite confusing. I suggest not to keep this file with sources, but put it directly to `dist`.

Ok, I get the point here. So for source it could return `unknown` hash. Actually, I changed it to `source` which should be meaningful in this case. So now `build.info.json` is generated, moved to dist and removed from `src` during build step. I also left `savehash` script intentionally (which only generates the file) so one can test/run it - it is also required for one CI job to be run before build, since build itself is done on Docker container with no `git` available. See https://github.com/scramjetorg/transform-hub/pull/402/commits/be8f980665b64985ace4fcaf4abfeca6355db84a.

Reply:
Thanks!